Cycle_Sort/Cycle_Sort.java

```diff
@@ -0,0 +1,79 @@
+/* 
+Cycle Sort
+Cycle sort is a comparison based sorting algorithm which forces array to be factored into 
+the number of cycles where each of them can be rotated to produce a sorted array.
+It is an in-place and unstable sorting algorithm. 
+Best and Average case time complexity:O(n^2)
+
+
+Algorithm
+An element a is given, index of a can be calculated by counting the number of elements that 
+are smaller than a.
+1. If the element is found to be at its correct position, simply leave it as it is.
+2. Otherwise, find the correct position of a by counting the total number of elements that are 
+less than a. where it must be present in the sorted array. The other element b which is replaced is 
+to be moved to its correct position. This process continues until we got an element at the 
+original position of a. */
+
+import java.io.*;
+import java.util.*;
+
+public class CycleSort
+{
+    static void sort (int a[], int n)
+    {
+        int writes = 0, start, element, pos, temp, i;
+        // Find position where we put the element.
+        for (start = 0; start <= n - 2; start++)
+        {
+            element = a[start];
+            pos = start;
+           /*Find position where we put the element. We basically 
+           count all smaller elements on right side of element. */
+           for (i = start + 1; i < n; i++)
+           if (a[i] < element)
+                pos++;
+        // If element is already in correct position     
```

Comment:
Indent these three comments as well. Then it will be perfect. And once you done with this, please squash your commits into a single commit.